homeworks/03-data-structure/src/BinarySearchTree.ts

```diff
@@ -0,0 +1,55 @@
+import { BinaryTreeI } from './BinaryTree';
+
+interface BinarySearchTreeI<T extends number | string> extends BinaryTreeI<T> {
+  acquire(value: T): boolean;
+}
+
+export class BinarySearchTree implements BinarySearchTreeI<number> {
```

Comment:
The naming for interface should be improved, adding suffix letter `I` doesn't look good instead what you can do is:
* name interface `BinarySearchTree`
* name class implementation `BinarySearchTreeImpl`